test/Telegram.Bot.Tests.Integ/Inline Mode/InlineQueryTests.cs

```diff
@@ -587,6 +587,68 @@ public InlineQueryTests(TestsFixture fixture)
             );
         }
 
+        [Fact(DisplayName = FactTitles.ShouldAnswerInlineQueryWithMpgeg4Gif)]
+        [Trait(Constants.MethodTraitName, Constants.TelegramBotApiMethods.AnswerInlineQuery)]
+        [ExecutionOrder(18)]
+        public async Task Should_Answer_Inline_Query_With_Mpeg4Gif()
+        {
+            await _fixture.SendTestCaseNotificationAsync(FactTitles.ShouldAnswerInlineQueryWithMpgeg4Gif,
+                startInlineQuery: true);
+
+            Update iqUpdate = await _fixture.UpdateReceiver.GetInlineQueryUpdateAsync();
+
+            InlineQueryResultBase[] results =
+            {
+                new InlineQueryResultMpeg4Gif(
+                    id: "mpeg4_gif_result",
+                    mpeg4Url: "https://pixabay.com/en/videos/download/video-10737_medium.mp4",
+                    thumbUrl: "https://i.vimeocdn.com/video/646283246_640x360.jpg"
+                )
+                {
+                    Caption = "A beatufiul scene",
+                },
+            };
+
+            await BotClient.AnswerInlineQueryAsync(
+                inlineQueryId: iqUpdate.InlineQuery.Id,
+                results: results,
+                cacheTime: 0
+            );
+        }
+
+        [Fact(DisplayName = FactTitles.ShouldAnswerInlineQueryWithCachedMpeg4Gif)]
+        [Trait(Constants.MethodTraitName, Constants.TelegramBotApiMethods.AnswerInlineQuery)]
+        [ExecutionOrder(19)]
+        public async Task Should_Answer_Inline_Query_With_Cached_Mpeg4Gif()
+        {
+            await _fixture.SendTestCaseNotificationAsync(FactTitles.ShouldAnswerInlineQueryWithCachedMpeg4Gif);
+
+            Message gifMessage = await BotClient.SendDocumentAsync(
+                chatId: _fixture.SupergroupChat,
+                document: "https://upload.wikimedia.org/wikipedia/commons/2/2c/Rotating_earth_%28large%29.gif",
```

Comment:
Apparently sending mp4 h264 encoded videos as document or video doesn't work with cached mpeg4_gif result. In both cases type mismatch error is returned.

Reply:
Interesting!